datatransfer/impl/graphsync/graphsync_impl.go

```diff
@@ -183,7 +195,8 @@ func (impl *graphsyncImpl) OpenPullDataChannel(ctx context.Context, requestTo pe
 	return chid, nil
 }
 
-// createNewChannel creates a new channel id and channel state and saves to channels
+// createNewChannel creates a new channel id and channel state and saves to channels.
+// It overwrites the existing value; it does not check if the map key exists already.
```

Comment:
this should be changed to check this and return an error if it exists. if initiator + tid is not enough to keep things unique, we're doing something wrong